base/compiler/ssair/inlining.jl

```diff
@@ -994,6 +994,22 @@ function process_simple!(ir::IRCode, idx::Int, params::OptimizationParams, world
     return (sig, invoke_data)
 end
 
+# This is not currently called in the regular course, but may be needed
+# if we ever want to re-run inlining again later in the pass pipeline after
+# additional type information was discovered.
+function recompute_method_matches(atype, sv)
+    # Regular case: Retrieve matching methods from cache (or compute them)
+    # World age does not need to be taken into account in the cache
+    # because it is forwarded from type inference through `sv.params`
```

Comment:
This seems confusing to state. The world age _is_ taken into account, because that's part of what `sv.params` means, and is partly why the cache is part of the params.

Reply:
That comment was there before ;)

Reply:
Though I should probably update it, because it's not the regular case anymore.